File: server/app/models/interview.py
from enum import Enum
from typing import List, Dict, Optional, Any
from pydantic import BaseModel
from datetime import datetime

class InterviewPhase(str, Enum):
    INTRODUCTION = "introduction"
    TECHNICAL = "technical"
    CRITICAL = "critical"
    CLOSING = "closing"

class ChatMessage(BaseModel):
    role: str
    content: str
    phase: InterviewPhase
    timestamp: datetime

class EvaluationScore(BaseModel):
    answer_quality: float
    technical_correctness: float
    communication: float
    problem_solving: float
    attitude_confidence: float
    overall_recommendation: float
    feedback_details: Dict[str, str]
    summary: str
    strengths: List[str]
    areas_for_improvement: List[str]

class InterviewState(BaseModel):
    phase: InterviewPhase = InterviewPhase.INTRODUCTION
    exchange_count: int = 0
    history: List[ChatMessage] = []
    resume_context: str = ""
    evaluation: Optional[EvaluationScore] = None

class InterviewSession(BaseModel):
    room_name: str
    candidate_name: str
    start_time: datetime
    end_time: Optional[datetime] = None
    state: InterviewState = InterviewState()
    resume_path: Optional[str] = None
    resume_parsed_text: Optional[str] = None
    report_pdf_path: Optional[str] = None
    vector_index: Any = None

File: client/src/types/interview.ts
export type InterviewPhase = 'introduction' | 'technical' | 'critical' | 'closing';

export interface EvaluationScore {
  answer_quality: number;
  technical_correctness: number;
  communication: number;
  problem_solving: number;
  attitude_confidence: number;
  overall_recommendation: number;
  feedback_details: Record<string, string>;
  summary: string;
  strengths: string[];
  areas_for_improvement: string[];
}

export interface SessionStatus {
  room_name: string;
  phase: InterviewPhase;
  exchange_count: number;
  report_ready: boolean;
}

export interface InterviewReport {
  score: EvaluationScore;
  pdfUrl: string;
}
